db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/subquery/exists/ExistsSubqueryFlatteningVisitor.java

```diff
@@ -370,7 +378,15 @@ private FromSubquery addFromSubquery(SelectNode outerSelectNode,
             /*
              * Left join outer tables with new FromSubquery.
              */
-            HalfOuterJoinNode outerJoinNode = nf.buildOuterJoinNode(outerSelectNode.getFromList(), fromSubquery, joinClause);
+            FromTable outerFromTable =
+                getFromTableContainingOuterColumnReference(outerSelectNode.getFromList(),
+                                                           outerColumnReference);
+            if (outerFromTable == null && outerSelectNode.getFromList().size() > 1) {
+                throw StandardException.newException(LANG_INTERNAL_ERROR,
```

Comment:
Is there an example, when it can happen, or it's just defense?

Reply:
This is mainly just defensive.  It is the job of getFromTableContainingOuterColumnReference to locate the outer table that outerColumnReference refers to, if there is more than one outer table (if there is just one, then we don't need to specifically locate the outer table, because there is just one).  If we can't locate the outer table, we don't know which one to remove from the WHERE clause so we need to error out.

Reply:
Hmm, now I see that we still assume `outerSelectNode.getFromList().size == 1`. Maybe we can duplicate or move the comment above here because now the check is actually outside of `buildOuterJoinNode`.

Reply:
Not sure which comment you're referring to.  I saw the following in buildOuterJoinNode:
   
> Currently we only attempt not-exist flattening if the outer table has one table.

Changed the comment to:

>         /* If outerFromTable is null, that means there is only one outer table.
>          * If outerFromTable is not null, it specifies the outer table we are flattening. 
>          */